dre-service/src/main/java/io/cdap/re/RulesDB.java

```diff
@@ -71,38 +72,39 @@
   private static final StructuredTableId RULES_TABLE_ID = new StructuredTableId("rules");
   static final StructuredTableSpecification RULES_TABLE_SPEC = new StructuredTableSpecification.Builder()
     .withId(RULES_TABLE_ID)
-    .withFields(new FieldType(ID_COL, FieldType.Type.STRING),
-                new FieldType(NAMESPACE_COL, FieldType.Type.STRING),
+    .withFields(new FieldType(NAMESPACE_COL, FieldType.Type.STRING),
+                new FieldType(ID_COL, FieldType.Type.STRING),
                 new FieldType(DESCRIPTION_COL, FieldType.Type.STRING),
                 new FieldType(CONDITION_COL, FieldType.Type.STRING),
                 new FieldType(ACTION_COL, FieldType.Type.STRING),
                 new FieldType(CREATED_COL, FieldType.Type.LONG),
                 new FieldType(UPDATED_COL, FieldType.Type.LONG))
-    .withPrimaryKeys(ID_COL)
+    .withPrimaryKeys(NAMESPACE_COL, ID_COL)
+    .withIndexes()
     .build();
   private static final StructuredTableId RULEBOOK_TABLE_ID = new StructuredTableId("rulebook");
   static final StructuredTableSpecification RULEBOOK_TABLE_SPEC = new StructuredTableSpecification.Builder()
     .withId(RULEBOOK_TABLE_ID)
-    .withFields(new FieldType(ID_COL, FieldType.Type.STRING),
-                new FieldType(NAMESPACE_COL, FieldType.Type.STRING),
+    .withFields(new FieldType(NAMESPACE_COL, FieldType.Type.STRING),
+                new FieldType(ID_COL, FieldType.Type.STRING),
                 new FieldType(DESCRIPTION_COL, FieldType.Type.STRING),
                 new FieldType(VERSION_COL, FieldType.Type.LONG),
                 new FieldType(SOURCE_COL, FieldType.Type.STRING),
                 new FieldType(USER_COL, FieldType.Type.STRING),
                 new FieldType(CREATED_COL, FieldType.Type.LONG),
                 new FieldType(UPDATED_COL, FieldType.Type.LONG),
                 new FieldType(RULES_COL, FieldType.Type.STRING))
-    .withPrimaryKeys(ID_COL)
+    .withPrimaryKeys(NAMESPACE_COL, ID_COL)
     .build();
   private static final String RULE_TEMPLATE = "\n" +
-    "  rule %s {\n" +
+    "  rule %s in namespace %s {\n" +
```

Comment:
namespace is not something that should show up in the rule. Namespace comes from the path in the REST endpoint, not from the rule template.

Reply:
Done